enhancements/sig-policy/98-long-term-compliance-history/README.md

```diff
@@ -0,0 +1,508 @@
+- parent_policy_standard - a standard name of the parent policy (e.g. `NIST SP 800-53`) to filter on. Multiple can be
+  provided with `?parent_policy_standard=NIST%20SP%20800-53&parent_policy_standard=NIST%20SP%20231-41` syntax.
+- policy - the name of the policy (e.g. a `ConfigurationPolicy`) that generated the compliance event to filter on.
+- policy_kind - the kind of the policy (e.g. `ConfigurationPolicy`) that generated the compliance event to filter on.
+- policy_api_group - the `apiGroup` of the policy (e.g. `policy.open-cluster-management.io`) that generated the
+  compliance event to filter on.
+- message_includes - a filter for compliance messages that include the following substring.
+- include_spec - a boolean to include the policy's spec in the return value. This is `false` by default.
+- before - an ISO-8601 timestamp to indicate only compliance events before this time should be shown.
+- after - an ISO-8601 timestamp to indicate only compliance events after this time should be shown.
+- page - the page number in the query. This defaults to 1.
+- per_page - the number of compliance events returned per page. This defaults to 20 and maxes out at 100.
+
+The output would look as follows and be sorted by timestamps in descending order. The content would be filtered based on
+the user's access as described above:
+
+```json
+{
+  "data": [
+    {
+      "cluster": {
+        "name": "cluster1"
+      },
+      "parent_policy": {
+        "name": "policies.etcd-encryption",
+        "catageories": ["CM Configuration Management"],
+        "controls": ["CM-2 Baseline Configuration"],
+        "standards": ["NIST SP 800-53"]
+      },
+      "policy": {
+        "apiGroup": "policy.open-cluster-management.io",
+        "kind": "ConfigurationPolicy",
+        "name": "etcd-encryption",
+        // Only shown if `include_spec=true`
+        "spec": {}
+      },
+      "event": {
+        "compliance": "NonCompliant",
+        "message": "configmaps [app-data] not found in namespace default",
+        "timestamp": "2023-07-19T18:25:43.511Z",
+        "metadata": {}
+      }
+    },
+    {
+      "cluster": {
+        "name": "cluster2"
+      },
+      "parent_policy": {
+        "name": "policies.etcd-encryption",
+        "catageories": [],
+        "controls": [],
+        "standards": []
+      },
+      "policy": {
+        "apiGroup": "policy.open-cluster-management.io",
+        "kind": "ConfigurationPolicy",
+        "name": "etcd-encryption",
+        "spec": {},
+        "specHash": {}
+      },
+      "event": {
+        "compliance": "Compliant",
+        "message": "configmaps [app-data] found as specified in namespace default",
+        "timestamp": "2023-07-19T18:25:41.523Z",
+        "metadata": {}
+      }
+    }
+  ],
+  "metadata": {
+    "page": 1,
+    "per_page": 20,
+    "total": 123
+  }
+}
+```
+
+##### Generating a Report
```

Comment:
This report might be very large, right? The proposed GET `/api/v1/compliance-events` has pagination, but a report might have more than 100 entries (the maximum page size). For one "report" style API I worked with (as a consumer), I think I had to do a POST with the parameters for the report, the API responded with some kind of Accepted response, and gave a report ID that could be used later in a GET when the report is ready.